src/providers/oauth2/mod.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  ManifestOAuth2ProviderSchema,
+  OAuth2ProviderOptions,
+} from "./types.ts";
+
+export const DefineOAuth2Provider = (
+  definition: OAuth2ProviderDefinitionArgs,
+): OAuth2Provider => {
+  return new OAuth2Provider(definition);
+};
+
+export enum OAuth2ProviderTypes {
+  CUSTOM = "CUSTOM",
+}
+
+export type OAuth2ProviderDefinitionArgs = {
+  /** A unique name for your provider */
+  provider_key: string;
+  /** Type of your provider */
+  provider_type: OAuth2ProviderTypes;
```

Comment:
Related to the `OAuth2ProviderTypes` enum, is there a way to conditionally accept different `OAuth2ProviderOptions` based on the `provider_type`? In the future, not all fields in `OAuth2ProviderTypes` will be required like they are today.

We can definitely do this validation in the App Manifest API layer, and make all fields optional here, but wanted to see if there's a way to do this in the SDK itself.